### PR TITLE
add pov-recovery unit tests and support for elastic scaling

### PR DESCRIPTION
**Pull request by alindima**
- unit tests for pov-recovery
- elastic scaling support (recovering multiple candidates in a single relay chain block)
- also some small cleanups
- also switches to candidates_pending_availability in `handle_empty_block_announce_data`

Fixes https://github.com/paritytech/polkadot-sdk/issues/3577

After https://github.com/paritytech/polkadot-sdk/pull/4097 is merged, we should also add a zombienet test, similar to the existing `0002-pov_recovery.toml` but which has a single collator using elastic scaling on multiple cores.

### CODE REVIEW

File: cumulus/client/network/src/lib.rs [hunk 1]
```diff
@@ -298,15 +298,20 @@ where
...
+			return Ok(Validation::Success { is_new_best: true })
+		}
+
+		let mut backed_blocks =
+			Self::backed_block_hashes(&relay_chain_interface, relay_chain_best_hash, para_id)
+				.await?;
+
+		let head_hash = HeadData(header.encode()).hash();
+
+		if backed_blocks.find(|block_hash| block_hash == &head_hash).is_some() {
```
Comment (bkchr) [inline cumulus/client/network/src/lib.rs:314]:
```suggestion
		if backed_blocks.any(|block_hash| block_hash == &head_hash) {
```

**Review by bkchr — APPROVED** [review]
Nice work :+1:

**Review by lexnv — APPROVED** [review]
LGTM! Im not super familiar with the code tho 👍

**Comment by alindima** [comment]
I added a version check for the parachain host runtime API so that we don't call the new function without it being available (on polkadot and kusama it wasn't yet upgraded)

**Review by skunert — APPROVED** [review]
Really Nice!

**Pull request merged**